Sources/Filters+Strings.swift

```diff
@@ -119,14 +123,31 @@ extension Filters {
       return escapeReservedKeywords(in: string)
     }
 
+    /// Removes newlines and other whitespace from a string. Takes an optional Mode argument:
+    ///   - all (default): remove all newlines and whitespaces
+    ///   - leading: remove newlines and only leading whitespaces
+    ///
+    /// - Parameters:
+    ///   - value: the value to be processed
+    ///   - arguments: the arguments to the function; expecting zero or one mode argument
+    /// - Returns: the trimmed string
+    /// - Throws: FilterError.invalidInputType if the value parameter isn't a string
     static func removeNewlines(_ value: Any?, arguments: [Any?]) throws -> Any? {
-      let removeSpaces = try Filters.parseBool(from: arguments, index: 0, required: false) ?? true
       guard let string = value as? String else { throw Filters.Error.invalidInputType }
+      let mode = try Filters.parseEnum(from: arguments, default: RemoveNewlinesModes.all)
 
-      let set: CharacterSet = removeSpaces ? .whitespacesAndNewlines : .newlines
-      let result = string.components(separatedBy: set).joined()
-
-      return result
+      switch mode {
+      case .all:
+        return string
+          .components(separatedBy: .whitespacesAndNewlines)
+          .joined()
+      case .leading:
+        return string
+          .components(separatedBy: .newlines)
+          .map { String($0.unicodeScalars.drop(while: { CharacterSet.whitespaces.contains($0) })) }
```

Comment:
Codebeat seems to be unhappy with nesting too deep here, maybe separate the closure used in that map in a private function to make it happy?